### PR TITLE
🍒[cxx-interop] Fix fallout of splitting libc++ module into many top-level modules

### PR DESCRIPTION
**Pull request by egorzhdan**
libc++ recently split the `std` module into many top-level modules: https://github.com/llvm/llvm-project/commit/571178a21a8bc105bf86cf4bf92f842e07792e1a. This caused substantial fallout for Swift's support for the C++ stdlib.

Original PRs:
* https://github.com/apple/swift/pull/70411
* https://github.com/apple/swift/pull/70445
* https://github.com/apple/swift/pull/70462
* https://github.com/apple/swift/pull/70466

rdar://119270491

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please test

**Review by DougGregor — APPROVED** [review]

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -7478,6 +7486,18 @@ bool importer::requiresCPlusPlus(const clang::Module *module) {
   });
 }
 
+bool importer::isCxxStdModule(const clang::Module *module) {
+  if (module->getTopLevelModuleName() == "std")
+    return true;
+  // In recent libc++ versions the module is split into multiple top-level
+  // modules (std_vector, std_utility, etc).
+  if (module->getTopLevelModule()->IsSystem &&
```
Comment (ributzka) [inline lib/ClangImporter/ClangImporter.cpp:7494]: Is the check for `IsSystem` required? The code doesn't do the same for just the `std` module.

Comment (egorzhdan) [reply lib/ClangImporter/ClangImporter.cpp:7494]: This check is meant to handle potential cases where a client has a non-stdlib module called, for instance, `std_extras`. It is not strictly required, but I think it reduces the possibility of treating a custom module as a stdlib module.

Comment (egorzhdan) [reply lib/ClangImporter/ClangImporter.cpp:7494]: Let me know if you think this is harmful, I can put up a separate patch to change this logic.

**Pull request merged**